src/app/api/advocates/search/route.ts

```diff
@@ -0,0 +1,41 @@
+import db from "../../../../db";
+import { advocates } from "../../../../db/schema";
+import { sql, ilike, or } from "drizzle-orm";
+
+export async function GET(request: Request) {
+    try {
+        const { searchParams } = new URL(request.url);
+        const q = searchParams.get("q")?.toLowerCase() ?? "";
+
+        if (!q) {
+            const data = await db.select().from(advocates);
+            return Response.json({ data });
+        }
+
+        const data = await db
+            .select()
+            .from(advocates)
+            .where(
+                or(
+                    ilike(advocates.firstName, `%${q}%`),
+                    ilike(advocates.lastName, `%${q}%`),
+                    ilike(advocates.city, `%${q}%`),
+                    ilike(advocates.degree, `%${q}%`),
+                    // sql`EXISTS (
```

Comment:
This is commented out because the jsonb column was breaking functionality.  Due to time constraints, I left this as is.  I will speak about this in the discussion.md file.